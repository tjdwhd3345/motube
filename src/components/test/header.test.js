import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Header from "../header/header";

describe("Header", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });

  // it("로고 클릭 시 /motube로 이동", () => {
  //   render(<Header />);

  //   const logo = screen.getAllByRole("img")[0];
  //   userEvent.click(logo);
  //   expect(location.pathname).toBe("/motube");
  // });

  it("input 입력 후 검색 확인", () => {
    const onSearchClick = jest.fn();

    render(<Header onSearchClick={onSearchClick} />);

    const input = screen.getByRole("searchbox");
    const button = screen.getByRole("button");
    userEvent.type(input, "bts");
    userEvent.click(button);

    expect(onSearchClick).toHaveBeenCalledTimes(1);
    expect(onSearchClick).toHaveBeenCalledWith("bts");
  });
});
