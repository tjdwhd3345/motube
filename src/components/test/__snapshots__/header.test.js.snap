// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="logoContainer"
    >
      <img
        alt="logo"
        class="logo"
        src="images/logo.png"
      />
      <span
        class="title"
      >
        MoTube
      </span>
    </div>
    <input
      class="searchInput"
      placeholder="검색"
      type="search"
      value=""
    />
    <button
      class="searchButton"
    >
      <img
        alt="search"
        src="images/search.png"
      />
    </button>
  </header>
</DocumentFragment>
`;
